### ISSUE TITLE
 end segment of circle arc scissoring

### ISSUE THREAD

**Issue by rollraw** [ocornut/imgui#4419]
version: 1.82
branch: master

at first, i noticed that with dx9 renderer strokes are always anti-aliased even if imgui anti-aliasing is disabled
using following code in a fresh imgui example:
```cpp
ImGui::GetStyle().AntiAliasedLines = false;
ImGui::GetStyle().AntiAliasedFill = false;
ImGui::GetStyle().AntiAliasedLinesUseTex = false;
ImGui::GetForegroundDrawList()->AddCircle({ 300, 600 }, 50.f, IM_COL32(243, 104, 224, 255), 12, 1.0f);
```
we will see this next with dx9:
![dx9](https://user-images.githubusercontent.com/15694466/128942900-1ee9ece1-1c0f-4199-9740-7e20226f7fcd.png)
and next with dx11:
![dx11](https://user-images.githubusercontent.com/15694466/128942957-15eea55c-98f8-49e8-9736-223a78b0f2ca.png)
i guess it could be related to bugs of dx9, because i tried to set dx9 renderstates to disable anti-aliasing but nothing changes

at second, circles (no sense filled/stroked) last segment looking scissored with magic '12' number as segments count (that will look for precached points), due to reduced segments count by 1 due to closed primitive drawing

**Comment by ocornut** [comment]
Thank you for the report.
There are two distinct issues there.

**With dx9 renderer strokes are always anti-aliased even if imgui anti-aliasing is disabled**

First question: even if it was fully the case, would you have an _issue_ with it? 
I did that
```
ImGui::GetStyle().AntiAliasedLines = ImGui::GetIO().KeyShift;
ImGui::GetStyle().AntiAliasedFill = ImGui::GetIO().KeyShift;
ImGui::GetStyle().AntiAliasedLinesUseTex = ImGui::GetIO().KeyShift;
```
![dx9_aa](https://user-images.githubusercontent.com/8225057/129011915-3aeaac32-de7f-4cf1-aa77-6d21040c4983.gif)
It doesn't look particularly "more" anti-aliased but I think the perception might vary depending on colors. 

However it is possible there is something else in your app, e.g. multi-sample AA done by the app or by the GPU driver could alter that perception.


**About the circle missing one segment:**
Confirmed. I'm quite confused as 89685b346c7b02b7d557678fd9d5d11a80967f83 (for 1.75) was explicitly intended to fix something and it appears it did the opposite. I think possibly the fix made sense before we switched to the new adaptive algorithm in 1.78 or a further change? Tagging @thedmd (or I'll check next time when I get back from holidays).

**Comment by thedmd** [comment]
@ocornut I will take a closer look at the missing circle segment.

**Comment by thedmd** [comment]
@ocornut It is fixed on PR. I think this is how it should be from the begging, not patching AddCircle was my oversight.

Commit and PR have an extra comment, which I hope make the change clear.

**Comment by rollraw** [comment]
> However it is possible there is something else in your app, e.g. multi-sample AA done by the app or by the GPU driver could alter that perception.
it's not a my own app, just imgui example from latest commit, i also tried to explicitely disable multisample AA with
```cpp
presentParams.MultiSampleType = D3DMULTISAMPLE_NONE;
presentParams.MultiSampleQuality = 0UL;
```
in the d3d device creation, but seems it doesnt helped, so yes could be related to gpu/drivers

**Comment by thedmd** [comment]
Is AA forced in driver settings?

**Comment by rollraw** [comment]
> Is AA forced in driver settings?

FXAA - off
Gamma - on
MSAA - 4x
MFAA - off

but does this makes sense when same code with dx11 looking different?

**Comment by ocornut** [comment]
Should be fixed by cb00972b878a0a4722c9214e6f7d46d97f092c3f
Thanks @thedmd !
